core/src/main/java/org/apache/iceberg/SnapshotParser.java

```diff
@@ -93,6 +95,16 @@ static void toJson(Snapshot snapshot, JsonGenerator generator) throws IOExceptio
       generator.writeNumberField(SCHEMA_ID, snapshot.schemaId());
     }
 
+    if (snapshot.manifestListKeyMetadata() != null) {
+      generator.writeStringField(MANIFEST_LIST_KEY_METADATA, snapshot.manifestListKeyMetadata());
+    }
+
+    // TODO discuss: do we need to sign the size value? Or sign the whole snapshot?
+    // Or rely on REST catalog? - the only option that prevents "full folder replacement" attack.
+    if (snapshot.manifestListSize() >= 0) {
```

Comment:
another small question here, we essentially are doing a transform here

manifestlists sizes < 0 become 0. 

Also nit: we are also ignoring 0's that get passed through although we will read this as 0 if it is missing.

Just wondering what the intent here is. I think it may be better to have a defined missing value? Not sure

Reply:
Yep, part of this thread https://github.com/apache/iceberg/pull/7770#discussion_r1546050438